src-bindings/vscode/vscode.ml

```diff
@@ -2215,6 +2215,14 @@ module TextDocumentContentProvider = struct
       [@@js.builder]]
 end
 
+module FileSystemWatcher = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val onDidChange : t -> callback:(Uri.t -> unit) -> Disposable.t [@@js.call]]
```

Comment:
You should use the `Event` functor here.

Reply:
Thanks, that makes sense indeed. I've addressed that in https://github.com/ocamllabs/vscode-ocaml-platform/pull/798/commits/0de141cff215a4f5170679e535a5fcadcc6ffe48